front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import store from "../../store.js";
+import { setDisplayingQuestion } from '../../store/analyzeTranscriptSlice.js';
+import IntentSearch from "../helpers/IntentSearch.js";
+
+let search = new IntentSearch();
+
+class IntentMenu extends Component {
+    
+    constructor(props) {
+        super(props);
+        this.ref = React.createRef();  // Create a React reference to the entire intent menu element
+        this.handleClickOutside = this.handleClickOutside.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.state = {
+            inputValue: "",
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside, false);
+    }
+    
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClickOutside, false);
+    }
+    
+    handleClickOutside(event) {
+        // Detect when the user clicks outside of the intent menu to hide it
+        // This works by checking whether the click target is contained by the intent menu element, which is identified through the ref created in the constructor. If not contained, then the click was outside of the menu and it should close.
+        if (this.ref.current && !this.ref.current.contains(event.target)) {
+          this.props.onClickOutside();
+        }
+    }
+
+    handleChange(event) {
+        // Update the input value state based on the search input field
+        this.setState({inputValue: event.target.value});
+    }
+
+    formatSearchResult = (text, slices) => {
+        // Return an element that displays and formats a slice of <text> based on <slices>
+        // Used to highlight the slice of an intent's question that matches the search string, and display the remaining text normally
+
+        // First, create slice boundaries to define when to change the text format
+        const startText = Math.max(slices[text].start - 5, 0);  // Display <text> starting a few chars before the searched portion
+        const startSearchSlice = slices[text].start;  // Receive the starting index of the searched portion
+        const endSearchSlice = slices[text].end;  // Receive the ending index of the searched portion
+
+        return(
+            <div className="truncate" /* Display <text> in 3 slices, with the middle one having highlighted formatting */>
+                {(startText > 0 ? "..." : "") + text.slice(startText, startSearchSlice) /* Begin text with ellipsis if preceding characters are truncated */}<span className='bg-gray-100'>{text.slice(startSearchSlice, endSearchSlice)}</span>{text.slice(endSearchSlice)}
```

Comment:
```suggestion
                {(startText > 0 ? "..." : "") + text.slice(startText, startSearchSlice) /* Begin text with ellipsis if preceding characters are truncated */}
                <span className='bg-gray-100'>
                    {text.slice(startSearchSlice, endSearchSlice)}
                </span>
                {text.slice(endSearchSlice)}
```

Reply:
rough idea to format nicer:
 put span on new line and content of span on new indented line

Reply:
wait I swear this did not work the first time I tried line breaking (it line broke the text on screen as well), which is why I asked on discord. But this works now! thanks

Reply:
np :D